JsonSubTypes/JsonSubtypes.cs

```diff
@@ -79,6 +79,8 @@ public KnownSubTypeWithPropertyAttribute(Type subType, string propertyName)
 
         [ThreadStatic] private static JsonReader _reader;
 
+        private static readonly Dictionary<TypeInfo, IEnumerable<object>> _attributesCache = new Dictionary<TypeInfo, IEnumerable<object>>();
```

Comment:
https://github.com/JamesNK/Newtonsoft.Json/issues/2414

Reply:
I agree that this adds small overhead on memory, but I think that size of this dictionary would be so small, that it will not change the whole thing much.

Reply:
Apparently for the .net4+ using a [`ConditionalWeakTable<TypeInfo, IEnumerable<object>>`](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.conditionalweaktable-2?view=netcore-3.1) would solve the issue pointed by @sungam3r 

Source: https://stackoverflow.com/questions/6601502/caching-reflection-data

The nice thing is that [it is thread safe](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.conditionalweaktable-2?redirectedfrom=MSDN&view=netcore-3.1#thread-safety)